orm/validator.py

```diff
@@ -477,6 +521,40 @@ def fsm_collision(one_fsm, two_fsm):
     return not one_fsm.isdisjoint(two_fsm)
 
 
+def validate_constraints_timeout_server(domain_rules=None, globals_doc=None):
+    # Assert that timeout_server is not out of bounds.
+    print("Checking for timeout_server out of bounds...")
+    for _, rules in domain_rules.items():
+        for rule in rules:
+            backend_config = rule.get("actions", {}).get("backend", {})
+            if backend_config:
+                haproxy = globals_doc.get("haproxy", {})
+                timeout_server_default = haproxy.get("timeout_server_default", 15000)
+                backend_timeout_server = backend_config.get(
+                    "timeout_server", timeout_server_default
+                )
+                timeout_server_max = haproxy.get("timeout_server_max", 15000)
+                if (
+                    timeout_server_max != 0
+                    and backend_timeout_server > timeout_server_max
+                ):
```

Comment:
What if `backend_timeout_server == 0` (infinite) and `timeout_server_max > 0`?

Reply:
uh oh